src/csharp/Microsoft.Spark/SparkContext.cs

```diff
@@ -289,6 +291,26 @@ public void AddFile(string path, bool recursive = false)
             _jvmObject.Invoke("addFile", path, recursive);
         }
 
+        /// <summary>
+        /// Add an archive to be downloaded and unpacked with this Spark job on every node.
+        /// </summary>
+        /// <remarks>
+        /// If an archive is added during execution, it will not be available until the next
+        /// TaskSet starts.
+        /// 
+        /// A path can be added only once. Subsequent additions of the same path are ignored.
+        /// </remarks>
+        /// <param name="path">
+        /// Archive path can be either a local file, a file in HDFS (or other Hadoop-supported
+        /// filesystems), or an HTTP, HTTPS or FTP URI. The given path should be one of .zip,
+        /// .tar, .tar.gz, .tgz and .jar.
+        /// </param>
+        [Since(Versions.V3_1_0)]
+        public void AddArchive(string path)
+        {
+            _jvmObject.Invoke("addArchive", path);
+        }
+
```

Comment:
Is there a reason we haven't included `[listArchives]`(https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/core/src/main/scala/org/apache/spark/SparkContext.scala#L1564)?

Reply:
I assumed `listFiles` was originally left out on purpose, so following the same pattern I left this out as well.  Although I don't really see any harm in adding both.

Reply:
Hmm, `listFiles` was probably never added because `pySpark` doesn't expose it.